src/lightning/pytorch/core/saving.py

```diff
@@ -87,7 +86,15 @@ def _load_from_checkpoint(
     if issubclass(cls, pl.LightningDataModule):
         return _load_state(cls, checkpoint, **kwargs)
     if issubclass(cls, pl.LightningModule):
-        return _load_state(cls, checkpoint, strict=strict, **kwargs)
+        if map_location is None:
+            map_location = list(checkpoint["state_dict"].values())[0].device
+        storage = _load_state(cls, checkpoint, strict=strict, **kwargs)
+        assert isinstance(storage, pl.LightningModule)
+        restore_location = torch.serialization._get_restore_location(map_location)
+        if isinstance(map_location, dict):
+            map_location = str(storage.device)
```

Comment:
Is this because `restore_location` does not support dict? Maybe you should send a patch for that to PyTorch

Reply:
Actually, because the torch.load handles it all fine we could simply do this
```python
if issubclass(cls, pl.LightningModule):
        map_location = list(checkpoint["state_dict"].values())[0].device
        storage = _load_state(cls, checkpoint, strict=strict, **kwargs)
        return storage.to(map_location)
```
and not even deal with the `torch.serialization._get_restore_location`, thus removing the `torch` import